encode_test.go

```diff
@@ -1104,19 +1104,24 @@ func TestWhitespaceWithContent(t *testing.T) {
 		},
 		{
 			"  hello  \n",
-			"\"  hello  \\n\"\n",
+			"|4\n      hello  \n",
 			"Multiple spaces + text + spaces + newline - should be double quoted (ends with spaces)",
 		},
 		{
 			"hello  \n",
-			"\"hello  \\n\"\n",
+			"|\n    hello  \n",
 			"Text + spaces + newline - should be double quoted (ends with spaces)",
```

Comment:
You talked about the change here

https://github.com/yaml/go-yaml/pull/79#issuecomment-3208952435

There is clear problem indeed

Not only the format changes (why not, after all) , but here the trailing spaces before the \n are lost

I feel like it's not a good thing.

Reply:
> Not only the format changes (why not, after all) , but here the trailing spaces before the \n are lost

I'm confused.
How are they lost?

Reply:
I'm sorry. I feel like my comment was based on a misunderstanding.

Now, I feel like I didn't understand clearly what your change is about.

The spaces are indeed there. I was confused by the presence of \n as a first character of the marshaled.

I feel like I didn't understand the change involved here. So I'll pass on this PR.